.github/cloud-samples-tools/cmd/main.go

```diff
@@ -0,0 +1,149 @@
+*/
+
+package main
+
+import (
+	"cloud-samples-tools/pkg/utils"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	command := ""
+	if len(os.Args) > 1 {
+		command = os.Args[1]
+	} else {
+		fmt.Fprintf(os.Stderr, "❌ no command specified\n")
+		printUsage(os.Stderr)
+		os.Exit(1)
+	}
+
+	switch command {
+	case "affected":
+		configFile := ""
+		if len(os.Args) > 2 {
+			configFile = os.Args[2]
+		} else {
+			fmt.Fprintf(os.Stderr, "❌ no config file specified\n")
+			printUsage(os.Stderr)
+			os.Exit(1)
+		}
+
+		diffsFile := ""
+		if len(os.Args) > 3 {
+			diffsFile = os.Args[3]
+		} else {
+			fmt.Fprintf(os.Stderr, "❌ no diffs file specified\n")
+			printUsage(os.Stderr)
+			os.Exit(1)
+		}
+
+		affectedCmd(configFile, diffsFile)
+
+	case "run-all":
+		configFile := ""
+		if len(os.Args) > 2 {
+			configFile = os.Args[2]
+		} else {
+			fmt.Fprintf(os.Stderr, "❌ no config file specified\n")
+			printUsage(os.Stderr)
+			os.Exit(1)
+		}
+
+		script := ""
+		if len(os.Args) > 3 {
+			script = os.Args[3]
+		} else {
+			fmt.Fprintf(os.Stderr, "❌ no script file specified\n")
+			printUsage(os.Stderr)
+			os.Exit(1)
+		}
+
+		runAllCmd(configFile, script)
+
+	default:
+		fmt.Fprintf(os.Stderr, "❌ unknown command: %s\n", command)
+		printUsage(os.Stderr)
+		os.Exit(1)
+	}
+}
+
+func printUsage(f *os.File) {
+	fmt.Fprintf(f, "usage: tools <command> ...\n")
+	fmt.Fprintf(f, "\n")
+	fmt.Fprintf(f, "commands:\n")
+	fmt.Fprintf(f, "  affected path/to/config.jsonc path/to/diffs.txt\n")
+	fmt.Fprintf(f, "  run-all path/to/config.jsonc path/to/script.sh\n")
+}
+
+func affectedCmd(configFile string, diffsFile string) {
+	config, err := utils.LoadConfig(configFile)
+	if err != nil {
+		log.Fatalf("❌ error loading the config file: %v\n%v\n", configFile, err)
+	}
+
+	diffsBytes, err := os.ReadFile(diffsFile)
+	if err != nil {
+		log.Fatalf("❌ error getting the diffs: %v\n%v\n", diffsFile, err)
+	}
+	diffs := strings.Split(string(diffsBytes), "\n")
+
+	packages, err := utils.Affected(config, diffs)
+	if err != nil {
+		log.Fatalf("❌ error finding the affected packages.\n%v\n", err)
+	}
+	if len(packages) > 256 {
+		log.Fatalf(
+			"❌ Error: GitHub Actions only supports up to 256 packages, got %v packages, for more details see:\n%v\n",
+			len(packages),
+			"https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow",
+		)
+	}
+
+	packagesJson, err := json.Marshal(packages)
+	if err != nil {
+		log.Fatalf("❌ error marshaling packages to JSON.\n%v\n", err)
+	}
+
+	fmt.Println(string(packagesJson))
+}
+
+func runAllCmd(configFile string, script string) {
+	config, err := utils.LoadConfig(configFile)
+	if err != nil {
+		log.Fatalf("❌ error loading the config file: %v\n%v\n", configFile, err)
+	}
+
+	packages, err := utils.FindAllPackages(".", config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "❌ error finding packages.\n%v\n", err)
+	}
+
+	maxGoroutines := 16
```

Comment:
nit: move this outside of the function as `const` in this file.

Reply:
Removed this function since it's currently unused.